module/zfs/zio_checksum.c

```diff
@@ -179,10 +283,12 @@ zio_checksum_compute(zio_t *zio, enum zio_checksum checksum,
 		else
 			bp->blk_cksum = eck->zec_cksum;
 		eck->zec_magic = ZEC_MAGIC;
-		ci->ci_func[0](data, size, &cksum);
+		ci->ci_func[0](data, size, spa->spa_cksum_tmpls[checksum],
```

Comment:
This is straight from https://github.com/openzfs/openzfs/commit/45818ee so I don't know if it's correct or not.  The code is still there in top-of-the-tree Illumos.